File: Python/Easy/HappyNumber.py
class Solution:
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        digits = set()
        while n != 1:
            n = sum(int(i) ** 2 for i in str(n))
            if n in digits:
                return False
            else:
                digits.add(n)
        return True

File: Java/Easy/HappyNumber.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer> loopedNumbers = new HashSet<Integer>();
        int square, remainder;
        while(loopedNumbers.add(n)){
            square = 0;
            while(n > 0){
                remainder = n % 10;
                square += remainder * remainder;
                n = n / 10;
            }
            if(square == 1)
                return true;
            else
                n = square;
        }
        return false;
    }
}